File: Easy/Nth Fibonacci Number/nth-fibonacci-number.py

class Solution:
    def nthFibonacci(self, n : int) -> int:
        # code here
        
        a=1
        b=1
        if n<=2:
            return 1
        while n-2:
            res = (a+b)%1000000007
            a =b
            b = res
            n-=1
        return res
        
        
        






#{ 
 # Driver Code Starts
if __name__=="__main__":
    t = int(input())
    for _ in range(t):
        
        n = int(input())
        
        obj = Solution()
        res = obj.nthFibonacci(n)
        
        print(res)
        

# } Driver Code Ends

File: Difficulty: Easy/Nth Fibonacci Number/nth-fibonacci-number.java
//{ Driver Code Starts
//Initial Template for Java
import java.util.*;
import java.io.*;
class GFG {
    public static void main(String args[]) throws IOException {
        BufferedReader read =
            new BufferedReader(new InputStreamReader(System.in));
        int t = Integer.parseInt(read.readLine());
        while (t-- > 0) {
            int n = Integer.parseInt(read.readLine());
            Solution ob = new Solution();

            System.out.println(ob.nthFibonacci(n));
        }
    }
}
// } Driver Code Ends


//User function Template for Java

//User function Template for Java
class Solution {
    static int nthFibonacci(int n){
        // code here
        if(n==1){
            return 0;
        }
        if(n==2){
            return 1;
        }
        int a =0, b=1;
        for(int i=2; i<=n; i++ ){
            int temp = a+b;
            a= b;
            b= temp%1000000007;
        }
        return b;
    }
}